bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -634,6 +639,11 @@ public Journal(File journalDirectory, ServerConfiguration conf, LedgerDirsManage
 
         this.removePagesFromCache = conf.getJournalRemovePagesFromCache();
         // read last log mark
+        if (conf.getJournalDirs().length == 1) {
+            lastMarkFileName = LAST_MARK_DEFAULT_NAME;
+        } else {
+            lastMarkFileName = LAST_MARK_DEFAULT_NAME + "." + journalIndex;
```

Comment:
What happens if the user changes the order of directories on the config file?

Reply:
I think the cookie stops it booting (but we should doublecheck)

Reply:
Yes the cookie is based on the list that comes from the config file, so changing the order will invalidate the cookie